### PR TITLE
[callsystem] CUTS Runner

### PR DESCRIPTION
**Pull request by daniwasonline**
# Description
This PR adds a special callsystem for running CUTS tests. It simply adds an otherwise-dull runner with a special wake command (`cuts.run [CUTS unit test ID or "all" for running all CUTS tests at once]`). When this wake command is called, the CUTS runner will ensure that the test ID is valid and isn't a regular callsystem. If these conditions are met, the runner will pass context to the CUTS test to complete an activation. Results are piped back to the runner and are sent as necessary.

# Why is this useful?
As CUTS unit tests are expressed as otherwise-regular callsystems that use special methods in the CUTS SDK/API, running multiple tests simultaneously or running multiple tests in fast succession is incredibly unfriendly, i.e. one may run into rate-limits. Having a special test runner callsystem will help ease this common scenario.

### CODE REVIEW

**Comment by Butterroach** [comment]
why am i supposed to review

**Comment by Butterroach** [comment]
idk what hte fuck is going on :sad:

File: src/callsystems/tests/cuts.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
+import { Callsystem } from "../../lib/callsystems/index.js";
+// from https://github.com/prathercc/date-to-discord-snowflake/blob/main/index.js
+const generateSnowflake = (date = new Date()) =>
```
Comment (artifishvr) [inline src/callsystems/tests/cuts.js:3]: unused code? 

```suggestion
 
```

File: src/callsystems/tests/cuts.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+	((BigInt(date.valueOf()) - BigInt(1420070400000)) << BigInt(22)).toString();
```
Comment (artifishvr) [inline src/callsystems/tests/cuts.js:4]:
```suggestion
	 
```

File: src/callsystems/tests/cuts.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
+import { Callsystem } from "../../lib/callsystems/index.js";
+// from https://github.com/prathercc/date-to-discord-snowflake/blob/main/index.js
```
Comment (artifishvr) [inline src/callsystems/tests/cuts.js:2]:
```suggestion
 
 ```

**Comment by artifishvr** [comment]
i am SO sorry i have no idea how to suggest multiple lines other than doing that

**Comment by artifishvr** [comment]
also bye butterroach :c

**Comment by Butterroach** [comment]
dont request reviews from me i know nothing about the new code base

**Comment by artifishvr** [comment]
its mildly humorous

**Comment by artifishvr** [comment]
> i know nothing about the new code base

yoooo me neitherrr (also i will stop)

**Comment by daniwasonline** [comment]
> i am SO sorry i have no idea how to suggest multiple lines other than doing that

its okay dw

**Comment by daniwasonline** [comment]
> > i know nothing about the new code base
> 
> yoooo me neitherrr (also i will stop)

i worked on it and i dont understand half of what i did (i do but i have a stroke every time i read it)

**Comment by daniwasonline** [comment]
time for crack

**Pull request merged**